mesonbuild/arglist.py

```diff
@@ -140,12 +140,15 @@ def flush_pre_post(self) -> None:
 
         #pre and post will overwrite every element that is in the container
         #only copy over args that are in _container but not in the post flush or pre flush set
+        if len(pre_flush_set) or len(post_flush_set):
```

Comment:
```suggestion
        if pre_flush_set or post_flush_set:
```

using `len()` to check the emptiness of a container is an anit-pattern, and much slower than just `if container`